internal/commands/fmt.go

```diff
@@ -42,10 +50,18 @@ func NewFormatCommand(ctx context.Context) *cobra.Command {
 					return fmt.Errorf("format: %w", err)
 				}
 
+				// If the original file contents match the formatted contents, formatting does not
+				// need to be done and we can try the next module.
 				if bytes.Equal(contents, formattedContents) {
 					continue
 				}
 
+				// When we are running the format command in check mode and the file contents are different
+				// we want to return an error code to the user and not update any of the files.
+				if viper.GetBool("check") {
+					os.Exit(1)
```

Comment:
I think we just want to return an error here and Cobra will take care of returning non-zero.

Reply:
It does, yeah -- but an error message will be returned rather than _just_ an error code. Did we want to include a message as well? e.g. we could just add

```
return errors.New("fmt required")
```

Reply:
I think to be of value we need to return the files that are not formatted correctly, otherwise folks will have to test each file independently to see which is causing the issue. So we'd have something like:

```go
// looping over each file ...
var unformattedFiles []string
if notMatch && check {
  unformattedFiles = append(unformattedFiles, file)
  continue
}
if notMatch {
// ... formatting code
}
// loop exits
if len(unformattedFiles) > 0 {
  return fmt.Errorf("files not formatted correctly: %v", unformattedFiles)
}
return nil
```

Reply:
I'm not entirely against the idea, but not quite sure if thats something an end user would really care about. My expected flow for this would be something like

```
conftest fmt policy --check (error 1 returned)
```

then they'd just run

```
conftest fmt policy 
```

locally and commit the change

So I just see it as a "ah ok, guess I need to run fmt"

Maybe go this route until theres a feature request for displaying the files?

Reply:
That works too. As long as we return an error rather than exiting here.

Reply:
@jalseth an error return has been added https://github.com/open-policy-agent/conftest/pull/655/files#diff-d637d8a600f1e167785fc077a5a8b196c8b83455e46a98bc3e576bc91714d6c5R63